### PR TITLE
#165413105 user profile

### PR DESCRIPTION
**Pull request by jabichris**
#### Description

This PR implements user profile functionalities, like editing username, bio or image. It allows also the user to view other users profiles.
#### Type of change

N/A
#### How has it been tested

It can be tested using the following link api/users/: id and use GET or PUT method in postman
#### Checklist:

N/A
#### Pivotal tracker story ID
https://www.pivotaltracker.com/story/show/165413105

### CODE REVIEW

**Review by Kabalisa — APPROVED** [review]
@jabichris Good work. remove the conflicts now.

**Review by Niyitangasam — APPROVED** [review]

File: tests/profile.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+chai.use(chaiHttp);
+
+describe('GET /api/users/:id', () => {
+  before(async () => {
+    const { User } = model;
+    try {
+      await User.create({
+        username: 'maningiri',
+        email: 'jabichris@gmail.com',
```
Comment (macphilips) [inline tests/profile.js]: Remove your email and put a dummy email there

File: tests/profile.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    chai
+      .request(app)
+      .get('/api/v1/users/1')
+      .end((err, res) => {
+        expect(res.status).to.equal(200);
+        expect(res.body.data).to.have.property('username');
+        expect(res.body.data).to.have.property('email');
+        expect(res.body.data).to.have.property('bio');
+        expect(res.body.data).to.have.property('image');
+        expect(res.body.data).to.have.property('password');
```
Comment (macphilips) [inline tests/profile.js]:
```suggestion
        expect(res.body.data).not.to.have.property('password');
```

File: routes/api/index.js [hunk 3]
```diff
@@ -19,7 +19,7 @@ router.use((err, req, res, next) => {
   }
   return next(err);
 });
-
+router.use('/users', users);
```
Comment (macphilips) [inline routes/api/index.js:22]:
```suggestion
router.use('/users', users, resetPassword);
```

File: routes/api/index.js [hunk 4]
```diff
@@ -19,7 +19,7 @@ router.use((err, req, res, next) => {
...
 router.use('/user', resetPassword);
```
Comment (macphilips) [inline routes/api/index.js]:
```suggestion

```

**Review by macphilips — CHANGES_REQUESTED** [review]

File: controllers/profile.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
+// import { now } from 'sequelize/types/lib/utils';
```
Comment (macphilips) [inline controllers/profile.js]: remove comment

**Pull request merged**